server/channels/app/platform/session.go

```diff
@@ -16,6 +16,15 @@ import (
 func (ps *PlatformService) ReturnSessionToPool(session *model.Session) {
 	if session != nil {
 		session.Id = ""
+		// Once the session is retrieved from the pool, all existing prop fields are cleared.
+		// To avoid a race between clearing the props and accessing it, clear the props maps before returning it to the pool.
+		for k := range session.Props {
+			delete(session.Props, k) // clear is only available in go 1.21
+		}
+		// Also clear the team members slice to avoid a similar race condition.
+		for i := 0; i < len(session.TeamMembers); i++ {
+			session.TeamMembers[i] = nil
+		}
```

Comment:
I'd expect the clearing logic to also reset the slice to 0 length.

Reply:
Agree. `session.TeamMembers = session.TeamMembers[:0]` would be ideal. And at this point, we are adding too much code detached from the session struct. I'd recommend refactoring this to a method and move all the code there.

Reply:
Actually, nvm. This is only for 9.5. It's okay to keep this code here. But we should definitely reset the slice length.

Reply:
Sorry, it looks like even `clear(session.TeamMembers)` does not reset the length of the slice. That seems wrong in master. We should fix that as well.

Reply:
Exactly, @agnivade. `clear` only resets every element to `nil`. This code does the same. Please note that the MsgPack code also only grows the slice and never shrinks it. (https://github.com/mattermost/mattermost/blob/cd970851a81ea01bf88558f9a4bf9ade04ee63ee/server/public/model/session_serial_gen.go#L380-L397)

I'm hesitant to change too much as part of my PRs that doesn't have to do with the original issues, a panic. Clearing the length to 0 means it has to grow every time it gets pulled from the pool. I'm not against making that change in general, but I don't think we should be making a change that potentially impacts performance in this PR.

Reply:
>  Clearing the length to 0 means it has to grow every time it gets pulled from the pool.

No. We are only setting the `length` to 0, not the `capacity`. The capacity remains the same as original.

I am okay if you don't want to change more than necessary in the PR. But then we shouldn't clear `TeamMembers` at all. Because if we do, then we should reset the slice length to 0 as well.

Reply:
Fair point. I've pushed 93ff9333d1, which reset the length to `0` while keeping the `capacity`. I will apply the same changes to https://github.com/mattermost/mattermost/pull/28254.